src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -58,23 +58,14 @@ private[spark] object Config {
   // See https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
   // Clients do not use the prefix, therefore we just accept the name portion on label selector names.
   private val label = "([\\w&&[^-_.]]([\\w-_.]{0,61}[\\w&&[^-_.]])?)"
-  private val labelSelectors: Regex = (s"^($label=$label(,$label=$label)*)?$$").r
+  private val labelSelectors: Regex = s"^($label=$label(,$label=$label)*)?$$".r
 
   private[armada] def selectorsValidator(selectors: CharSequence): Boolean = {
-    val selectorsMaybe = labelSelectors.findPrefixMatchOf(selectors)
-    selectorsMaybe match {
-      case Some(_) => true
-      case None => false
-    }
+    labelSelectors.findPrefixMatchOf(selectors).isDefined
   }
 
   def transformSelectorsToMap(str: String): Map[String,String] = {
-    if (str.trim.isEmpty) {
-      Map()
-    }
-    else {
-      str.split(",").map(a => a.split("=")(0) -> a.split("=")(1)).toMap
-    }
+    str.trim.split(",").map(a => a.split("=", 2)).map(a => a(0) -> a(1)).toMap
```

Comment:
Doesn't this throw an exception if the string is empty?:
```
scala> var str = ""
scala>     str.trim.split(",").map(a => a.split("=", 2)).map(a => a(0) -> a(1)).toMap
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
```

Reply:
How about this instead:
```
str.trim.split(",").toSeq.filter(_ != "").map(a => a.split("=", 2)).map(a => a(0) -> a(1)).toMap
```

Reply:
Almost,  `!_.isEmpty`  can be simplified to `_.nonEmpty`, and after the `split` we have to filter for `_.length == 2`. And of course, there could be more inner `trim`s:
```
    str.trim.split(",").filter(_.nonEmpty)
      .map(a => a.trim.split("=", 2))
      .filter(_.length == 2)
      .map(_.map(_.trim))
      .map(a => a(0) -> a(1))
      .toMap
```